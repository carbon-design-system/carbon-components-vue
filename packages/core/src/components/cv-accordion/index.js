import CvAccordion from './cv-accordion';
import CvAccordionItem from './cv-accordion-item';
import CvAccordionSkeleton from './cv-accordion-skeleton';

export { CvAccordion, CvAccordionItem, CvAccordionSkeleton };
