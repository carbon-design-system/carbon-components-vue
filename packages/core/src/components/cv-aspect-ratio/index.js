import CvAspectRatio from './cv-aspect-ratio';
export { CvAspectRatio };
