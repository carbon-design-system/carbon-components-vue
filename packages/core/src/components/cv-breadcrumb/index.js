import CvBreadcrumb from './cv-breadcrumb';
import CvBreadcrumbItem from './cv-breadcrumb-item';
import CvBreadcrumbSkeleton from './cv-breadcrumb-skeleton';

export { CvBreadcrumb, CvBreadcrumbItem, CvBreadcrumbSkeleton };
