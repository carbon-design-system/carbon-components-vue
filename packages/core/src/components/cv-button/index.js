import CvButton from './cv-button';
import CvIconButton from './cv-icon-button';
import CvButtonSkeleton from './cv-button-skeleton';
import CvButtonSet from './cv-button-set';

export { CvButton, CvIconButton, CvButtonSkeleton, CvButtonSet };
