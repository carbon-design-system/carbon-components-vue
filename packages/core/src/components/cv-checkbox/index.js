import CvCheckbox from './cv-checkbox';
import CvCheckboxSkeleton from './cv-checkbox-skeleton';
export { CvCheckbox, CvCheckboxSkeleton };
