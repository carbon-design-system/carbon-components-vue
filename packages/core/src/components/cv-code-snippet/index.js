import CvCodeSnippet from './cv-code-snippet';
import CvCodeSnippetSkeleton from './cv-code-snippet-skeleton';
export { CvCodeSnippet, CvCodeSnippetSkeleton };
