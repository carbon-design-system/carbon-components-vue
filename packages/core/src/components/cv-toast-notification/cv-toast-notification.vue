<template>
  <div
    data-notification
    :class="[
      `cv-notifiation ${carbonPrefix}--toast-notification`,
      `${carbonPrefix}--toast-notification--${kind.toLowerCase()}`,
      { [`${carbonPrefix}--toast-notification--low-contrast`]: lowContrast },
    ]"
    v-on="$listeners"
    :role="isAlert ? 'alert' : undefined"
    :aria-live="!isAlert ? 'polite' : false"
  >
    <component :is="icon" :class="`${carbonPrefix}--toast-notification__icon`" />
    <div :class="`${carbonPrefix}--toast-notification__details`">
      <h3 :class="`${carbonPrefix}--toast-notification__title`">{{ title }}</h3>
      <p :class="`${carbonPrefix}--toast-notification__subtitle`" v-html="subTitle"></p>
      <p :class="`${carbonPrefix}--toast-notification__caption`" v-html="caption"></p>
    </div>
    <button
      :aria-label="closeAriaLabel"
      type="button"
      data-notification-btn
      :class="`${carbonPrefix}--toast-notification__close-button`"
      @click="$emit('close')"
    >
      <Close20 :class="`${carbonPrefix}--toast-notification__close-icon`" />
    </button>
  </div>
</template>

<script>
import { notificationMixin, carbonPrefixMixin } from '../../mixins';
import ErrorFilled20 from '@carbon/icons-vue/es/error--filled/20';
import CheckmarkFilled20 from '@carbon/icons-vue/es/checkmark--filled/20';
import WarningFilled20 from '@carbon/icons-vue/es/warning--filled/20';
import Close20 from '@carbon/icons-vue/es/close/20';
import InformationFilled20 from '@carbon/icons-vue/es/information--filled/20';

export default {
  name: 'CvToastNotification',
  components: { Close20 },
  mixins: [notificationMixin, carbonPrefixMixin],
  props: {
    caption: String,
    closeAriaLabel: { type: String, default: 'Dismiss notification' },
    kind: {
      type: String,
      default: 'info',
      validator: val => ['error', 'info', 'warning', 'success'].includes(val),
    },
    lowContrast: Boolean,
  },
  computed: {
    icon() {
      switch (this.kind) {
        case 'error':
          return ErrorFilled20;
        case 'warning':
          return WarningFilled20;
        case 'success':
          return CheckmarkFilled20;
        case 'info':
          return InformationFilled20;
        default:
          return '';
      }
    },
  },
};
</script>
