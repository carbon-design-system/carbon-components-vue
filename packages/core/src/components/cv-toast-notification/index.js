import CvToastNotification from './cv-toast-notification';
export { CvToastNotification };
