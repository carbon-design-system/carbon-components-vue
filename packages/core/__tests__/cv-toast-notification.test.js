// import { shallowMount as shallow } from '@vue/test-utils';
import { testComponent, awaitNextTick } from './_helpers';
const { shallowMount: shallow, trigger } = awaitNextTick;
import { CvToastNotification } from '@/components/cv-toast-notification';
import ErrorFilled20 from '@carbon/icons-vue/es/error--filled/20';
import CheckmarkFilled20 from '@carbon/icons-vue/es/checkmark--filled/20';
import WarningFilled20 from '@carbon/icons-vue/es/warning--filled/20';
import InformationFilled20 from '@carbon/icons-vue/es/information--filled/20';

describe('CvToastNotification', () => {
  const kinds = ['error', 'info', 'warning', 'success'];

  // ***************
  // PROP CHECKS
  // ***************
  testComponent.propsAreType(CvToastNotification, ['lowContrast', 'hideCloseButton'], Boolean);
  testComponent.propsAreType(CvToastNotification, ['caption', 'closeAriaLabel', 'kind'], String);
  testComponent.propsHaveDefault(CvToastNotification, ['closeAriaLabel', 'kind']);

  it('`kind` prop validator works as expected', async () => {
    const propsData = { caption: 'TEST', lowContrast: false };

    const wrapper = await shallow(CvToastNotification, { propsData });
    for (const kind of kinds) {
      expect(wrapper.vm.$options.props.kind.validator && wrapper.vm.$options.props.kind.validator(kind)).toBeTruthy();
    }

    // suppress the error message from the kind validator
    const consoleError = console.error;
    console.error = jest.fn();

    expect(wrapper.vm.$options.props.kind.validator && wrapper.vm.$options.props.kind.validator('TEST')).toBeFalsy();

    // restore
    console.error = consoleError;
  });

  // ***************
  // SNAPSHOT TESTS
  // ***************

  it('should render correctly', async () => {
    const propsData = { caption: 'TEST', lowContrast: false, kind: '' };
    for (const kind of kinds) {
      propsData.kind = kind;
      const wrapper = await shallow(CvToastNotification, { propsData });
      expect(wrapper.html()).toMatchSnapshot();
    }
  });

  it('should render correctly when low contrast is used', async () => {
    const propsData = { caption: 'TEST', lowContrast: true, kind: '' };
    for (const kind of kinds) {
      propsData.kind = kind;
      const wrapper = await shallow(CvToastNotification, { propsData });
      expect(wrapper.html()).toMatchSnapshot();
    }
  });

  it('should not render close button when hideCloseButton is true', async () => {
    const propsData = { lowContrast: false, kind: '', actionLabel: 'test action label', hideCloseButton: true };
    for (const kind of kinds) {
      propsData.kind = kind;
      const wrapper = await shallow(CvToastNotification, { propsData });
      expect(wrapper.html()).toMatchSnapshot();
    }
  });

  // ***************
  // FUNCTIONAL TESTS
  // ***************

  it('`icon` is computed correctly', async () => {
    const kindToIconMapping = {
      error: ErrorFilled20,
      warning: WarningFilled20,
      success: CheckmarkFilled20,
      info: InformationFilled20,
    };
    const propsData = { caption: 'TEST', lowContrast: false, kind: '' };
    for (const kind of kinds) {
      propsData.kind = kind;
      const wrapper = await shallow(CvToastNotification, { propsData });
      expect(wrapper.vm.icon).toEqual(kindToIconMapping[kind]);
    }
  });

  it('should emit close event when close button is clicked', async () => {
    const propsData = { caption: 'TEST', lowContrast: false };
    const wrapper = await shallow(CvToastNotification, { propsData });
    await trigger(wrapper.find('button'), 'click');
    expect(wrapper.emitted().close).toBeTruthy();
  });
});
