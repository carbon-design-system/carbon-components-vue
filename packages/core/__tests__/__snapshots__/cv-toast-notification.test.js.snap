// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CvToastNotification should not render close button when hideCloseButton is true 1`] = `
<div data-notification="" role="alert" class="cv-notification bx--toast-notification bx--toast-notification--error" actionlabel="test action label">
  <anonymous-stub class="bx--toast-notification__icon"></anonymous-stub>
  <div class="bx--toast-notification__details">
    <h3 class="bx--toast-notification__title"></h3>
    <p class="bx--toast-notification__subtitle"></p>
    <p class="bx--toast-notification__caption"></p>
  </div>
  <!---->
</div>
`;

exports[`CvToastNotification should not render close button when hideCloseButton is true 2`] = `
<div data-notification="" aria-live="polite" class="cv-notification bx--toast-notification bx--toast-notification--info" actionlabel="test action label">
  <anonymous-stub class="bx--toast-notification__icon"></anonymous-stub>
  <div class="bx--toast-notification__details">
    <h3 class="bx--toast-notification__title"></h3>
    <p class="bx--toast-notification__subtitle"></p>
    <p class="bx--toast-notification__caption"></p>
  </div>
  <!---->
</div>
`;

exports[`CvToastNotification should not render close button when hideCloseButton is true 3`] = `
<div data-notification="" role="alert" class="cv-notification bx--toast-notification bx--toast-notification--warning" actionlabel="test action label">
  <anonymous-stub class="bx--toast-notification__icon"></anonymous-stub>
  <div class="bx--toast-notification__details">
    <h3 class="bx--toast-notification__title"></h3>
    <p class="bx--toast-notification__subtitle"></p>
    <p class="bx--toast-notification__caption"></p>
  </div>
  <!---->
</div>
`;

exports[`CvToastNotification should not render close button when hideCloseButton is true 4`] = `
<div data-notification="" aria-live="polite" class="cv-notification bx--toast-notification bx--toast-notification--success" actionlabel="test action label">
  <anonymous-stub class="bx--toast-notification__icon"></anonymous-stub>
  <div class="bx--toast-notification__details">
    <h3 class="bx--toast-notification__title"></h3>
    <p class="bx--toast-notification__subtitle"></p>
    <p class="bx--toast-notification__caption"></p>
  </div>
  <!---->
</div>
`;

exports[`CvToastNotification should render correctly 1`] = `
<div data-notification="" role="alert" class="cv-notification bx--toast-notification bx--toast-notification--error">
  <anonymous-stub class="bx--toast-notification__icon"></anonymous-stub>
  <div class="bx--toast-notification__details">
    <h3 class="bx--toast-notification__title"></h3>
    <p class="bx--toast-notification__subtitle"></p>
    <p class="bx--toast-notification__caption">TEST</p>
  </div> <button aria-label="Dismiss notification" type="button" data-notification-btn="" class="bx--toast-notification__close-button">
    <close20-stub class="bx--toast-notification__close-icon"></close20-stub>
  </button>
</div>
`;

exports[`CvToastNotification should render correctly 2`] = `
<div data-notification="" aria-live="polite" class="cv-notification bx--toast-notification bx--toast-notification--info">
  <anonymous-stub class="bx--toast-notification__icon"></anonymous-stub>
  <div class="bx--toast-notification__details">
    <h3 class="bx--toast-notification__title"></h3>
    <p class="bx--toast-notification__subtitle"></p>
    <p class="bx--toast-notification__caption">TEST</p>
  </div> <button aria-label="Dismiss notification" type="button" data-notification-btn="" class="bx--toast-notification__close-button">
    <close20-stub class="bx--toast-notification__close-icon"></close20-stub>
  </button>
</div>
`;

exports[`CvToastNotification should render correctly 3`] = `
<div data-notification="" role="alert" class="cv-notification bx--toast-notification bx--toast-notification--warning">
  <anonymous-stub class="bx--toast-notification__icon"></anonymous-stub>
  <div class="bx--toast-notification__details">
    <h3 class="bx--toast-notification__title"></h3>
    <p class="bx--toast-notification__subtitle"></p>
    <p class="bx--toast-notification__caption">TEST</p>
  </div> <button aria-label="Dismiss notification" type="button" data-notification-btn="" class="bx--toast-notification__close-button">
    <close20-stub class="bx--toast-notification__close-icon"></close20-stub>
  </button>
</div>
`;

exports[`CvToastNotification should render correctly 4`] = `
<div data-notification="" aria-live="polite" class="cv-notification bx--toast-notification bx--toast-notification--success">
  <anonymous-stub class="bx--toast-notification__icon"></anonymous-stub>
  <div class="bx--toast-notification__details">
    <h3 class="bx--toast-notification__title"></h3>
    <p class="bx--toast-notification__subtitle"></p>
    <p class="bx--toast-notification__caption">TEST</p>
  </div> <button aria-label="Dismiss notification" type="button" data-notification-btn="" class="bx--toast-notification__close-button">
    <close20-stub class="bx--toast-notification__close-icon"></close20-stub>
  </button>
</div>
`;

exports[`CvToastNotification should render correctly when low contrast is used 1`] = `
<div data-notification="" role="alert" class="cv-notification bx--toast-notification bx--toast-notification--error bx--toast-notification--low-contrast">
  <anonymous-stub class="bx--toast-notification__icon"></anonymous-stub>
  <div class="bx--toast-notification__details">
    <h3 class="bx--toast-notification__title"></h3>
    <p class="bx--toast-notification__subtitle"></p>
    <p class="bx--toast-notification__caption">TEST</p>
  </div> <button aria-label="Dismiss notification" type="button" data-notification-btn="" class="bx--toast-notification__close-button">
    <close20-stub class="bx--toast-notification__close-icon"></close20-stub>
  </button>
</div>
`;

exports[`CvToastNotification should render correctly when low contrast is used 2`] = `
<div data-notification="" aria-live="polite" class="cv-notification bx--toast-notification bx--toast-notification--info bx--toast-notification--low-contrast">
  <anonymous-stub class="bx--toast-notification__icon"></anonymous-stub>
  <div class="bx--toast-notification__details">
    <h3 class="bx--toast-notification__title"></h3>
    <p class="bx--toast-notification__subtitle"></p>
    <p class="bx--toast-notification__caption">TEST</p>
  </div> <button aria-label="Dismiss notification" type="button" data-notification-btn="" class="bx--toast-notification__close-button">
    <close20-stub class="bx--toast-notification__close-icon"></close20-stub>
  </button>
</div>
`;

exports[`CvToastNotification should render correctly when low contrast is used 3`] = `
<div data-notification="" role="alert" class="cv-notification bx--toast-notification bx--toast-notification--warning bx--toast-notification--low-contrast">
  <anonymous-stub class="bx--toast-notification__icon"></anonymous-stub>
  <div class="bx--toast-notification__details">
    <h3 class="bx--toast-notification__title"></h3>
    <p class="bx--toast-notification__subtitle"></p>
    <p class="bx--toast-notification__caption">TEST</p>
  </div> <button aria-label="Dismiss notification" type="button" data-notification-btn="" class="bx--toast-notification__close-button">
    <close20-stub class="bx--toast-notification__close-icon"></close20-stub>
  </button>
</div>
`;

exports[`CvToastNotification should render correctly when low contrast is used 4`] = `
<div data-notification="" aria-live="polite" class="cv-notification bx--toast-notification bx--toast-notification--success bx--toast-notification--low-contrast">
  <anonymous-stub class="bx--toast-notification__icon"></anonymous-stub>
  <div class="bx--toast-notification__details">
    <h3 class="bx--toast-notification__title"></h3>
    <p class="bx--toast-notification__subtitle"></p>
    <p class="bx--toast-notification__caption">TEST</p>
  </div> <button aria-label="Dismiss notification" type="button" data-notification-btn="" class="bx--toast-notification__close-button">
    <close20-stub class="bx--toast-notification__close-icon"></close20-stub>
  </button>
</div>
`;
